src/libultra/io/pfsinitpak.c

```diff
@@ -1,5 +1,126 @@
+#include "PR/pfs.h"
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/pfsinitpak/osPfsInitPak.s")
+s32 __osPfsCheckRamArea(OSPfs* pfs);
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/pfsinitpak/__osPfsCheckRamArea.s")
+s32 osPfsInitPak(OSMesgQueue* queue, OSPfs* pfs, s32 channel) {
+    s32 ret;
+    u16 sum;
+    u16 isum;
+    u8 temp[BLOCKSIZE];
+    __OSPackId* id;
+    __OSPackId newid;
+
+    __osSiGetAccess();
+
+    ret = __osPfsGetStatus(queue, channel);
+
+    __osSiRelAccess();
+
+    if (ret != 0) {
+        return ret;
+    }
+
+    pfs->queue = queue;
+    pfs->channel = channel;
+    pfs->status = 0;
+
+    ret = __osPfsCheckRamArea(pfs);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = __osPfsSelectBank(pfs, 0);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = __osContRamRead(pfs->queue, pfs->channel, PFS_ID_0AREA, temp);
+    if (ret != 0) {
+        return (ret);
+    }
+
+    __osIdCheckSum((u16*)temp, &sum, &isum);
+    id = (__OSPackId*)temp;
+    if ((id->checksum != sum) || (id->invertedChecksum != isum)) {
+        ret = __osCheckPackId(pfs, id);
+        if (ret != 0) {
+            pfs->status |= PFS_ID_BROKEN;
+            return ret;
+        }
+    }
+
+    if ((id->deviceid & 0x01) == 0) {
```

Comment:
Based on how this is assigned in `__osRepairPackId` I think this makes sense
```suggestion
    if (!(id->deviceid & 1)) {
```